### ISSUE TITLE
Need a Baker Developer for Custom Job

### ISSUE THREAD

**Issue by kvirella** [Simbul/baker#834]
Hi! I've used Baker and Laker to build a magazine that I want to sell both on my website and via Newsstand. After readers buy a subscription on my website, I'll need to be able to authenticate them in the Newsstand app. I purchased an off-the-shelf software program that authenticates the users but need help integrating it. The software program is written in PHP and uses a mySQL database. It comes with a REST API. Please let me know if any of you developers are interested in taking on this project.

Thanks!
Kelly

**Comment by nin9creative** [comment]
This is a good question/point to raise.  Many current Newsstand applications have some sort of method to enable a subscription that was purchased or is applicable as part of perhaps an existing print subscription.

This is something that I’ve thought about with regards to MagRocket - however it’s a bigger effort that would require Baker changes as well as a backend method for adding subscriptions purchased outside of Newsstand.

Not a simple task.

However a one off implementation might not be as hard to do for your specific instance if you already have something that manages the purchases.  You’d basically need to make some changes in the Baker Framework that would allow the user to circumvent the Newsstand related bits if a user has bought a susbscription.  However that breaks a lot of the current Baker which assumes Newsstand.

Why not just direct the user to the app to purchase?

Andrew

**Comment by folletto** [comment]
Hello @kvirella, I think it might be helpful to you to check out page with the list of developers that provide this kind of service: http://bakerframework.com/developers :)

**Comment by nin9creative** [comment]
@foletto... Kelly does bring up a good point though regarding the feature of being able to provide access to a subscription purchased outside of the Newsstand App.

Offer Content to New and Existing Subscribers

If you offer auto-renewable subscriptions, you can also use other methods to acquire digital subscribers outside of your app. You can sell digital subscriptions on your website or provide free access to content for existing subscribers. In these cases there is no revenue sharing since Apple was not involved in the transactions. Developers keep 100% of the revenue. If you would like to make a subscription offer outside of the app, the same (or better) subscription price must be offered inside the app for users who wish to subscribe from within the app. In addition, you may not provide links in your apps which allow the customer to purchase content or subscriptions outside of the app.

**Comment by folletto** [comment]
Absolutely. But that requires an even more specialized backend, most of the work is there really to handle all the exceptions, security, subscription, payment and so on. Baker can do little on that side and I'm not even sure the complexity of it, since already for subscriptions it's "overriding" Newsstand with the custom server subscription setup. In a sense, it's more a feature request for MagRocket. ;)

We'll see in the future. :)

**Comment by nin9creative** [comment]
I know :)  lol  I guess that it's certainly a point for consideration in the future!

Andrew

**Comment by kvirella** [comment]
@nin9creative Newsstand is easier, but as you noted Apple takes 30 percent of transactions and they take about 60 days to remit payments. Neither of those are great terms and now that Apple has stopped requiring publishers to sell their non-Newsstand subscriptions for the same price or more, adding this functionality seems like a win-win to us. Let me know if you do any custom development and would be interested. I'm very interested in MagRocket by the way. I did a lot of research on the market for such a solution last year. If you're looking for customer feedback, I'd be glad to chat with you.

@folletto Thanks for he tip about the baker developer page. I've also reached out to all of those.
